featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -518,25 +575,35 @@ def _build_transform_features(self, all_features, entity, max_depth=0,
                                                         new_max_depth,
                                                         input_types,
                                                         groupby_prim)
-            # get IDs to use as groupby
-            id_matches = self._features_by_type(all_features=all_features,
-                                                entity=entity,
-                                                max_depth=new_max_depth,
-                                                variable_type=set([Id]))
+
+            matching_inputs = filter_matches_by_options(matching_inputs, current_options)
+            # get columns to use as groupbys, use IDs as default unless other groupbys specified
+            if any(['include_groupby_variables' in option for option in current_options]):
+                default_type = variable_types.PandasTypes._all
+            else:
+                default_type = set([Id])
+            groupby_matches = self._features_by_type(all_features=all_features,
+                                                     entity=entity,
+                                                     max_depth=new_max_depth,
+                                                     variable_type=default_type)
+            # Convert groupby matches to tuples to make them consistent with inputs
+            groupby_matches = [(groupby_match, ) for groupby_match in groupby_matches]
+            groupby_matches = filter_matches_by_options(groupby_matches, current_options, groupby=True)
```

Comment:
i think it makes more sense to have two methods `filter_groupby_matches_by_options` and `filter_matches_by_options`.

Knowing to use the `groupby` parameter is no different than knowing 2 different methods exist. Also, then we can move the `[(groupby_match, ) for groupby_match in groupby_matches]` into the function rather than outside it.